game-core/src/main/java/games/strategy/triplea/ai/proAI/ProScrambleAi.java

```diff
@@ -126,15 +127,16 @@
 
     // Return units to scramble
     final HashMap<Territory, Collection<Unit>> scrambleMap = new HashMap<>();
-    for (final Territory t : possibleScramblers.keySet()) {
-      for (final Unit u : possibleScramblers.get(t).getSecond()) {
+    for (final Map.Entry<Territory, Tuple<Collection<Unit>, Collection<Unit>>> territoryTupleEntry : possibleScramblers
+        .entrySet()) {
+      for (final Unit u : territoryTupleEntry.getValue().getSecond()) {
         if (unitsToScramble.contains(u)) {
-          if (scrambleMap.containsKey(t)) {
-            scrambleMap.get(t).add(u);
+          if (scrambleMap.containsKey(territoryTupleEntry.getKey())) {
+            scrambleMap.get(territoryTupleEntry.getKey()).add(u);
```

Comment:
Interesting: There's no use of the value in this for loop, this might be a false positive